qa/tasks/cephfs/test_admin.py

```diff
@@ -1424,15 +1424,49 @@ def test_multiple_path_rw(self):
 
         self._remount_and_run_tests(FS_AUTH_CAPS, keyring)
 
+    def test_when_MDS_caps_needs_update_but_others_dont(self):
+        '''
+        Test that the command "ceph fs authorize" successfully updates MDS
+        caps even when MON and OSD caps don't need an update.
+
+        Tests: https://tracker.ceph.com/issues/64182
+
+        In this test we run -
+
+            ceph fs authorize cephfs2 client.x / rw
+            ceph fs authorize cephfs2 client.x /dir1 rw
+            ceph fs authorize cephfs2 client.x /dir2 rw
+
+        The first command will create the keyring by adding the MDS cap for
+        root path & MON and OSD caps with name of the FS name (say "cephfs1").
+        But second and third command don't need to add MON and OSD caps since
+        cap for "cephfs2" has been granted already. Thus, second and third
+        command only need to update the MDS cap, thus testing the bug
+        under consideration here.
+        '''
+        PERM = 'rw'
+        FS_AUTH_CAPS = (('/', PERM), ('dir1', PERM), ('/dir2', PERM))
+        self.captesters = []
+        for cap in FS_AUTH_CAPS:
+            if cap[0] != '/':
+                self.mount_a.run_shell(f'mkdir -p .{cap[0]}')
+            self.captesters.append(CapTester(self.mount_a, cap[0]))
+
+        for caps in FS_AUTH_CAPS:
+            keyring = self.fs.authorize(self.client_id, caps)
+
+        self._remount_and_run_tests(FS_AUTH_CAPS, keyring)
+
     def _remount_and_run_tests(self, fs_auth_caps, keyring):
         for i, c in enumerate(fs_auth_caps):
             self.assertIn(i, (0, 1))
             PATH = c[0]
             PERM = c[1]
             self._remount(keyring, PATH)
             # actual tests...
-            self.captesters[i].run_cap_tests(self.fs, self.client_id, PERM,
-                                             PATH)
+            self.captesters[i].run_cap_tests(self.fs, self.client_id, PERM)
+            #self.captesters[i].run_cap_tests(self.fs, self.client_id, PERM,
```

Comment:
why is this commented out?

Reply:
This part wasn't tested. Needed to take care of it eventually.

Reply:
Sure, please remove it if its unneeded.

Reply:
Done locally. I'll update the PR shortly.

Reply:
Done. Marking this thread resolved.